lib/pavilion/test_run.py

```diff
@@ -923,6 +929,80 @@ def create_id_dir(id_dir):
     def __repr__(self):
         return "TestRun({s.name}-{s.id})".format(s=self)
 
+    def _get_skipped(self):
+        match_list = []
+        match_list = self._match(match_list)
+
+        if len(match_list) is 0:
+            return False
+        else:
+            self.status.set(STATES.COMPLETE, match_list[0])
+            return True
+
+    def _make_regex(self, cond):
+        """_make_regex takes a dictionary and checks each value
+        under all keys and appends/prepends [^$] to make sure all
+        values are compatible with the regex match for conditional
+        statements. This is needed if the user is not using regex.
+        It explicitly defines the given value and only matches to it.
+        :param dict: Dictionary containing conditional statements
+        :return dict: Dictionary with updated values."""
+        cond_dict = self.config.get(cond, {})
+        for key in cond_dict:
+            list = []
+            for value in cond_dict[key]:
+                if len(value) is 0:
+                    value = '^$'
+                if value[0] is '^':
+                    pass
+                else:
+                    value = ''.join(['^', value])
+                if value[len(value)-1] is '$':
+                    pass
+                else:
+                    value = ''.join([value, '$'])
+                list.append(value)
+            cond_dict.update({key: list})
+        return cond_dict
+
+    def _match(self, match_list):
+
+        """Match grabs conditional keys from the config. It checks for
+        matches and depending on the results will skip or continue a test.
+        :param match_list: Match list is a list of conditional matches found.
+        :return The match list after being populated
+        :rtype list(String)"""
+        from pavilion.output import dbg_print
+        var_man = self.var_man
+        only_if = self._make_regex('only_if')
+        not_if = self._make_regex('not_if')
+
+        for nkey in not_if:
+            var_set, var, idx, sub_var = var_man.resolve_key(nkey)
+            if var_man.is_deferred(var_set, var, idx=idx, sub_var=sub_var):
+                continue
+            for vals in not_if[nkey]:
```

Comment:
The name `vals` is plural and implies that this variable is a list, when it's actually a string.